### PR TITLE
Fix node tool cleanup

### PR DESCRIPTION
**Pull request by henningandersen**
In case of failures, node tool could end up removing up both old
and new cluster state.

### CODE REVIEW

**Comment by henningandersen** [comment]
@elasticmachine run elasticsearch-ci/2

**Review by andrershov — CHANGES_REQUESTED** [review]
@henningandersen It would be nice to add a test case for this change - old metadata cleanup fails, but the tool execution is successful and we can read new metadata. Looks good otherwise.

**Comment by henningandersen** [comment]
>  It would be nice to add a test case for this change - old metadata cleanup fails, but the tool execution is successful and we can read new metadata.

Added in ed3812ce7b64c1021439fc282114e1f530cbc701

Thanks @andrershov

**Review by andrershov — CHANGES_REQUESTED** [review]
@henningandersen thank you for adding a test for this. I think that instead of adding a custom method that looks for all metadata files, we can use `MetaDataStateFormat.loadLatestStateWithGeneration` and `MetaDataStateFormat.loadGeneration` for `Manifest` and `Metadata` to check that the first run of the tool will not delete new generation. And the next run of the tool will delete both previous generations and create a new one.

**Comment by henningandersen** [comment]
> we can use `MetaDataStateFormat.loadLatestStateWithGeneration` and `MetaDataStateFormat.loadGeneration` for `Manifest` and `Metadata` to check that the first run of the tool will not delete new generation.

Thanks @andrershov , fixed in 9ff3b35df4442c4507ebfdb7013487535790d628, please have another look.

File: server/src/test/java/org/elasticsearch/cluster/coordination/ElasticsearchNodeCommandIT.java [hunk 1]
```diff
@@ -418,4 +423,57 @@ public void testCanRunUnsafeBootstrapAfterErroneousDetachWithoutLoosingMetaData(
...
+        final Manifest secondManifest = loadLatestManifest(dataPaths, namedXContentRegistry);
+        final MetaData secondMetaData = loadMetaData(dataPaths, namedXContentRegistry, secondManifest);
+        assertThat(secondManifest.getGlobalGeneration(), greaterThan(originalManifest.getGlobalGeneration()));
+        assertNotEquals(originalMetaData.clusterUUID(), secondMetaData.clusterUUID());
+
+        // check that a new run will cleanup.
+        executeCommand(new UnsafeBootstrapMasterCommand(), environment, 0, false);
+
+        assertNull(loadMetaData(dataPaths, namedXContentRegistry, originalManifest));
+        assertNull(loadMetaData(dataPaths, namedXContentRegistry, secondManifest));
```
Comment (andrershov) [inline server/src/test/java/org/elasticsearch/cluster/coordination/ElasticsearchNodeCommandIT.java]: Can we also check that loadLatestManifest and loadMetadata will load non-null objects?

**Review by andrershov — APPROVED** [review]
LGTM. I left one small comment, after it's addressed, no need for the next round.

File: server/src/test/java/org/elasticsearch/cluster/coordination/ElasticsearchNodeCommandIT.java (hunk 1 above)
Comment (henningandersen) [reply server/src/test/java/org/elasticsearch/cluster/coordination/ElasticsearchNodeCommandIT.java]: Fixed in ca59377ea745834b2d1477a1cde5d62cfc68fe77

**Pull request merged**